agent/api/task/task_linux.go

```diff
@@ -178,8 +182,14 @@ func (task *Task) buildImplicitLinuxCPUSpec() specs.LinuxCPU {
 	// aggregate container CPU shares when present
 	var taskCPUShares uint64
 	for _, container := range task.Containers {
-		if container.CPU < minimumCPUShare {
-			taskCPUShares += minimumCPUShare
+		if container.CPU < minimumCPUShares {
+			taskCPUShares += minimumCPUShares
+		} else if container.CPU > maximumCPUShares {
+			// If one of the containers is requesting more than the max permitted CPU shares,
+			// then we set taskCPUShares to be the maximumCPUShares and break out of this loop.
+			// This is to prevent the taskCPUShares to be more than the max CPU shares permitted by the Linux kernel.
+			taskCPUShares = maximumCPUShares
+			break
 		} else {
 			taskCPUShares += uint64(container.CPU)
```

Comment:
Another idea of putting a cap could be, we aggregate and then put a cap. Say we have 2 containers each with container.CPU = `200000`. Then aggregated task cpu shares could be out of bounds, if I understand the problem we have now?

Reply:
> Say we have 2 containers each with container.CPU = 200000. Then aggregated task cpu shares could be out of bounds, if I understand the problem we have now?

Yes, the sum would be 400000, and hence out of bounds.

Reply:
let me think about this. you're suggestion is valid

Reply:
right, the example you shared isnt solved by the current code. I'll make the changes, thanks!!

Reply:
updated! and also added this example as a unit test case, thanks!